pandas/core/reshape/pivot.py

```diff
@@ -734,3 +749,57 @@ def _get_names(arrs, names, prefix: str = "row"):
             names = list(names)
 
     return names
+
+
+def _build_names_mapper(
+    names: List[str], shared_col_row_names: Set[str], suffix: str
+) -> Tuple[Dict[str, str], List[str]]:
+    """
+    Given a list of row or column names, creates a mapper of unique names to
+    column/row names.
+
+    Parameters
+    ----------
+    names : list
```

Comment:
instead of creating uniques names, can we simply map input names -> position (which by definition are unique); still returning a dict and renaming is all ok.

Reply:
We can. My only issue with that if there are any errors when creating the DataFrame (line 618) or calling df.pivot_table (line 628) then the user would see unrecognizable column names for those errors.